core/src/executor/evaluate/function.rs

```diff
@@ -105,6 +105,22 @@ pub fn lower(name: String, expr: Evaluated<'_>) -> Result<Evaluated> {
     )))
 }
 
+pub fn initcap(name: String, expr: Evaluated<'_>) -> Result<Evaluated> {
+    let string = eval_to_str!(name, expr);
+    let mut char_vector = Vec::<char>::new();
+    let mut prev_char_alphanum = false;
+    for c in string.chars() {
+        if prev_char_alphanum {
+            char_vector.push(c.to_ascii_lowercase());
+        } else {
+            char_vector.push(c.to_ascii_uppercase());
+        }
+        prev_char_alphanum = c.is_alphanumeric();
+    }
+    let string = char_vector.iter().collect::<String>();
+    Ok(Evaluated::from(Value::Str(string)))
```

Comment:
We are trying to keep convetion to avoid `mut` keyword in `core` module.
Looks like we can try to refactor like below. 
(though `scan` use internal state as mut ref)
```suggestion
let string = string
    .chars()
    .scan(false, |prev_char_alphanum, current| {
        let char = match prev_char_alphanum {
            true => current.to_ascii_lowercase(),
            false => current.to_ascii_uppercase(),
        };
        *prev_char_alphanum = current.is_alphanumeric();

        Some(char)
    })
    .collect::<String>();

Ok(Evaluated::from(Value::Str(string)))
```

Reply:
Thanks